web3/process.go

```diff
@@ -119,90 +138,137 @@ func (c *Contracts) StateRoot(processID []byte) (*types.BigInt, error) {
 // the process. It returns the transaction hash of the state transition
 // submission, or an error if the submission fails. The tx hash can be used to
 // track the status of the transaction on the blockchain.
-func (c *Contracts) SetProcessTransition(processID, proof, inputs []byte, blobsSidecar *gtypes.BlobTxSidecar) (*common.Hash, error) {
+func (c *Contracts) SetProcessTransition(processID, proof, inputs []byte, blobsSidecar *gtypes.BlobTxSidecar) ([]byte, *common.Hash, error) {
+	// If the transaction manager is not available, return an error
+	if c.txManager == nil {
+		return nil, nil, fmt.Errorf("transaction manager not initialized")
+	}
+	// Copy processID into a fixed-size array to match the contract's expected
+	// type
 	var pid [32]byte
 	copy(pid[:], processID)
-	ctx, cancel := context.WithTimeout(context.Background(), web3QueryTimeout)
+	ctx, cancel := context.WithTimeout(context.Background(), web3WaitTimeout)
 	defer cancel()
-	if blobsSidecar == nil { // Regular transaction if no blobs are provided
-		autOpts, err := c.authTransactOpts()
-		if err != nil {
-			return nil, fmt.Errorf("failed to create transact options: %w", err)
-		}
-		autOpts.Context = ctx
-		tx, err := c.processes.SubmitStateTransition(autOpts, pid, proof, inputs)
-		if err != nil {
-			return nil, fmt.Errorf("failed to submit state transition: %w", err)
-		}
-		hash := tx.Hash()
-		return &hash, nil
-	}
-
+	// Prepare the ABI for packing the data
 	processABI, err := c.ProcessRegistryABI()
 	if err != nil {
-		return nil, fmt.Errorf("failed to get process registry ABI: %w", err)
+		return nil, nil, fmt.Errorf("failed to get process registry ABI: %w", err)
 	}
-	tx, err := c.NewEIP4844Transaction(
-		ctx,
-		c.ContractsAddresses.ProcessRegistry,
-		processABI,
-		"submitStateTransition",
-		[]any{pid, proof, inputs},
-		blobsSidecar,
-	)
-	if err != nil {
-		return nil, fmt.Errorf("failed to create EIP-4844 transaction: %w", err)
+	// Use transaction manager for automatic nonce management
+	var sentTx *gtypes.Transaction
+	txID, txHash, err := c.txManager.SendTx(ctx, func(nonce uint64) (*gtypes.Transaction, error) {
+		internalCtx, cancel := context.WithTimeout(context.Background(), web3WaitTimeout)
+		defer cancel()
+
+		var tx *gtypes.Transaction
+		if blobsSidecar == nil {
```

Comment:
I think this code snippet would be much human-friendly and neat with a switch-case. Do not "recycle" the error check, just perform all operations depending on the case. It will be cleaner.